File: ezSCORES/ezSCORES.Model/Competitions.cs
using ezSCORES.Model.ENUMs;
using System;
using System.Collections.Generic;
using System.Text;

namespace ezSCORES.Model
{
	public class Competitions
	{
		public int Id { get; set; }
		public int UserId { get; set; }

		public int SelectionId { get; set; }

		public string Season { get; set; } = null!;

		public int CityId { get; set; }

		public string Name { get; set; } = null!;

		public string Description { get; set; } = null!;

		public CompetitionType CompetitionType { get; set; }

		public int MaxTeamCount { get; set; }

		public byte[]? Picture { get; set; }

		public DateTime StartDate { get; set; }

		public DateTime ApplicationEndDate { get; set; }

		public int? Fee { get; set; }
		public CompetitionStatus Status { get; set; } = ENUMs.CompetitionStatus.Preparation;

		public int MaxPlayersPerTeam { get; set; }
		public virtual ICollection<CompetitionsReferees> CompetitionsReferees { get; set; } = new List<CompetitionsReferees>();
		public virtual ICollection<CompetitionsSponsors> CompetitionsSponsors { get; set; } = new List<CompetitionsSponsors>();
		public virtual ICollection<Rewards> Rewards { get; set; } = new List<Rewards>();
		public virtual ICollection<Reviews> Reviews { get; set; } = new List<Reviews>();

		public virtual Selections Selection { get; set; } = null!;
		public virtual Cities City { get; set; } = null!;
	}
}

File: ezSCORES/UI/ezscores_desktop/lib/models/competitions.dart
import 'package:ezscores_desktop/models/cities.dart';
import 'package:ezscores_desktop/models/competitionsReferees.dart';
import 'package:ezscores_desktop/models/competitionsSponsors.dart';
import 'package:ezscores_desktop/models/enums/competitionStatus.dart';
import 'package:ezscores_desktop/models/enums/competitionType.dart';
import 'package:ezscores_desktop/models/enums/competitions_status_converter.dart';
import 'package:ezscores_desktop/models/enums/competitions_type_converter.dart';
import 'package:ezscores_desktop/models/reviews.dart';
import 'package:ezscores_desktop/models/rewards.dart';
import 'package:ezscores_desktop/models/selections.dart';
import 'package:json_annotation/json_annotation.dart';

part 'competitions.g.dart';

@JsonSerializable(explicitToJson: true)
@CompetitionTypeConverter()
@CompetitionStatusConverter()
class Competitions {
	int? id;
	int? userId;
	int? selectionId;
	String? season;
	int? cityId;
	String? name;
	String? description;
	int? maxTeamCount;
	String? picture;
	DateTime? startDate;
	DateTime? applicationEndDate;
	int? fee;
	int? maxPlayersPerTeam;
  CompetitionType? competitionType;
  CompetitionStatus? status;
  Cities? city;
  Selections? selection;
  List<CompetitionsReferees>? competitionsReferees;
  List<CompetitionsSponsors>? competitionsSponsors;
  List<Rewards>? rewards;
  List<Reviews>? reviews;
 
	Competitions({
    this.id,
    this.userId,
    this.selectionId,
    this.season,
    this.cityId,
    this.name,
    this.description,
    this.maxTeamCount,
    this.picture,
    this.startDate,
    this.applicationEndDate,
    this.fee,
    this.maxPlayersPerTeam,
    this.competitionType,
    this.status,
    this.city,
    this.selection,
    this.competitionsReferees,
    this.competitionsSponsors,
    this.rewards,
    this.reviews,
  });

	factory Competitions.fromJson(Map<String, dynamic> json) => _$CompetitionsFromJson(json);

	Map<String, dynamic> toJson() => _$CompetitionsToJson(this);
}